diff -up gdm-2.20.5/gui/gdmuser.c.cap-user-list-length gdm-2.20.5/gui/gdmuser.c
--- gdm-2.20.5/gui/gdmuser.c.cap-user-list-length	2008-06-25 11:48:02.000000000 -0400
+++ gdm-2.20.5/gui/gdmuser.c	2008-06-25 13:55:15.000000000 -0400
@@ -266,7 +266,7 @@ setup_user (struct passwd *pwent,
 	    gboolean read_faces)
 {
     GdmUser *user;
-    int cnt = 0;
+    static int cnt = 0;
 
     if (pwent->pw_shell && 
 	gdm_check_shell (pwent->pw_shell) &&
@@ -296,11 +296,6 @@ setup_user (struct passwd *pwent,
 	    }
 
 	    if (cnt > 1000 || time_started + 5 <= time (NULL)) {
-		*users = g_list_append (*users,
-			g_strdup (_("Too many users to list here...")));
-		*users_string = g_list_append (*users_string,
-			g_strdup (_("Too many users to list here...")));
-
 		return (FALSE);
 	    }
     }
