diff -up gdm-2.26.1/daemon/gdm-session-direct.c.set-auth-flag gdm-2.26.1/daemon/gdm-session-direct.c
--- gdm-2.26.1/daemon/gdm-session-direct.c.set-auth-flag	2009-05-08 11:43:58.935662919 -0400
+++ gdm-2.26.1/daemon/gdm-session-direct.c	2009-05-08 11:46:46.785623788 -0400
@@ -380,6 +380,8 @@ gdm_session_direct_handle_authenticated 
         dbus_connection_send (conversation->worker_connection, reply, NULL);
         dbus_message_unref (reply);
 
+        session->priv->is_authenticated = TRUE;
+
         _gdm_session_authenticated (GDM_SESSION (session), conversation->service_name);
 
         return DBUS_HANDLER_RESULT_HANDLED;
@@ -407,6 +409,8 @@ gdm_session_direct_handle_authentication
 
         g_debug ("GdmSessionDirect: Emitting 'authentication-failed' signal");
 
+        session->priv->is_authenticated = FALSE;
+
         _gdm_session_authentication_failed (GDM_SESSION (session), conversation->service_name, NULL);
 
         return DBUS_HANDLER_RESULT_HANDLED;
