diff -up gdm-2.28.1/gui/simple-greeter/plugins/password/gdm-password-extension.c.eviction-notice gdm-2.28.1/gui/simple-greeter/plugins/password/gdm-password-extension.c
--- gdm-2.28.1/gui/simple-greeter/plugins/password/gdm-password-extension.c.eviction-notice	2009-11-03 20:27:15.162323893 -0500
+++ gdm-2.28.1/gui/simple-greeter/plugins/password/gdm-password-extension.c	2009-11-03 20:27:43.172313652 -0500
@@ -308,7 +308,6 @@ create_actions (GdmPasswordExtension *ex
                                  _("Log In"), NULL, NULL);
         g_signal_connect_swapped (action, "activate",
                                   G_CALLBACK (on_activate_log_in), extension);
-        g_object_set (G_OBJECT (action), "icon-name", "go-home", NULL);
         gtk_action_group_add_action (extension->priv->actions,
                                      action);
 }
diff -up gdm-2.28.1/gui/simple-greeter/plugins/smartcard/gdm-smartcard-extension.c.eviction-notice gdm-2.28.1/gui/simple-greeter/plugins/smartcard/gdm-smartcard-extension.c
--- gdm-2.28.1/gui/simple-greeter/plugins/smartcard/gdm-smartcard-extension.c.eviction-notice	2009-11-03 20:29:37.987313316 -0500
+++ gdm-2.28.1/gui/simple-greeter/plugins/smartcard/gdm-smartcard-extension.c	2009-11-03 20:28:00.438313043 -0500
@@ -439,7 +439,6 @@ create_actions (GdmSmartcardExtension *e
                                  _("Log In"), NULL, NULL);
         g_signal_connect_swapped (action, "activate",
                                   G_CALLBACK (on_activate_log_in), extension);
-        g_object_set (G_OBJECT (action), "icon-name", "go-home", NULL);
         gtk_action_group_add_action (extension->priv->actions,
                                      action);
 
diff -up gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.glade.eviction-notice gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.glade
--- gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.glade.eviction-notice	2009-11-05 12:46:06.072238114 -0500
+++ gdm-2.28.0/gui/simple-greeter/gdm-greeter-login-window.glade	2009-11-05 12:46:14.695237567 -0500
@@ -559,7 +559,6 @@
 			  <child>
 			    <widget class="GtkImage" id="image1">
 			      <property name="visible">True</property>
-			      <property name="icon_name">go-home</property>
 			      <property name="pixel_size">16</property>
 			      <property name="xalign">0.5</property>
 			      <property name="yalign">0.5</property>
