diff -up gdm-2.20.2/gui/gdmsetup.desktop.in.in.gdmsetup-path gdm-2.20.2/gui/gdmsetup.desktop.in.in
--- gdm-2.20.2/gui/gdmsetup.desktop.in.in.gdmsetup-path	2007-11-26 18:11:04.000000000 -0500
+++ gdm-2.20.2/gui/gdmsetup.desktop.in.in	2007-11-27 01:29:31.000000000 -0500
@@ -2,8 +2,8 @@
 Encoding=UTF-8
 _Name=Login Window
 _Comment=Configure GDM login window appearance and behavior
-TryExec=@sbindir@/gdmsetup
-Exec=@sbindir@/gdmsetup
+TryExec=/usr/bin/gdmsetup
+Exec=/usr/bin/gdmsetup
 Icon=gdmsetup
 StartupNotify=true
 Terminal=false
