diff -up gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in.in.registryd-no-wrapper gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in.in
--- gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in.in.registryd-no-wrapper	2009-10-05 23:34:13.222648781 -0400
+++ gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in.in	2009-10-05 23:35:09.702413116 -0400
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
 _Name=AT SPI Registry Wrapper
-Exec=@LIBDIR@/gnome-session/helpers/at-spi-registryd-wrapper
+Exec=/usr/libexec/at-spi-registryd
 OnlyShowIn=GNOME;
 AutostartCondition=GNOME /desktop/gnome/interface/accessibility
 X-GNOME-Autostart-Phase=Initialization
-X-GNOME-AutoRestart=true
\ No newline at end of file
+X-GNOME-AutoRestart=true
diff -up gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in.registryd-no-wrapper gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in
--- gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in.registryd-no-wrapper	2009-10-05 23:35:38.035399160 -0400
+++ gdm-2.28.0/data/greeter-autostart/at-spi-registryd-wrapper.desktop.in	2009-10-05 23:36:13.691664097 -0400
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
 _Name=AT SPI Registry Wrapper
-Exec=/usr/lib/gnome-session/helpers/at-spi-registryd-wrapper
+Exec=/usr/libexec/at-spi-registryd
 OnlyShowIn=GNOME;
 AutostartCondition=GNOME /desktop/gnome/interface/accessibility
 X-GNOME-Autostart-Phase=Initialization
-X-GNOME-AutoRestart=true
\ No newline at end of file
+X-GNOME-AutoRestart=true
